encode.go

```diff
@@ -40,6 +40,9 @@ type TableEncoder struct {
 	// lineStyle is the table line style.
 	lineStyle LineStyle
 
+	// rowStyles is the table of row style.
+	rowStyles map[[4]rune]rowStyle
```

Comment:
This should probably not be a map, but a struct, like `lineStyle` above, just to make the calls shorter, that is `enc.rowStyles.Mid` instead of `enc.rowStyles[enc.lineStyle.Mid]`